sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3495,6 +3495,24 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     assert(df4.schema.head.name === "randn(1)")
     checkIfSeedExistsInExplain(df2)
   }
+
+  test("SPARK-31710: " +
+    "TIMESTAMP_SECONDS, TIMESTAMP_MILLISECONDS and TIMESTAMP_MICROSECONDS to timestamp transfer") {
+    val df1 = sql("select TIMESTAMP_SECONDS(1230219000)," +
+      "TIMESTAMP_SECONDS(-1230219000),TIMESTAMP_SECONDS(null)")
+    checkAnswer(df1, Row(Timestamp.valueOf("2008-12-25 07:30:00"),
+      Timestamp.valueOf("1931-01-07 00:30:00"), null))
+
+    val df2 = sql("select TIMESTAMP_MILLISECONDS(1230219000000)," +
+      "TIMESTAMP_MILLISECONDS(-1230219000000),TIMESTAMP_MILLISECONDS(null)")
+    checkAnswer(df2, Row(Timestamp.valueOf("2008-12-25 07:30:00"),
+      Timestamp.valueOf("1931-01-07 00:30:00"), null))
+
+    val df3 = sql("select TIMESTAMP_MICROSECONDS(1230219000000000)," +
+      "TIMESTAMP_MICROSECONDS(-1230219000000000),TIMESTAMP_MICROSECONDS(null)")
+    checkAnswer(df3, Row(Timestamp.valueOf("2008-12-25 07:30:00"),
+      Timestamp.valueOf("1931-01-07 00:30:00"), null))
```

Comment:
I have added these sqls to `datetime.sql`, but not sure `datetime.sql` will check the correctness of these functions,so these checks is not removed currently, could you please give me a help and decide if these should retain? @cloud-fan

Reply:
the `datetime.sql` does check the correctness, isn't it? https://github.com/apache/spark/pull/28534/files#diff-79dd276be45ede6f34e24ad7005b0a7cR26

Reply:
@cloud-fan `datetime.sql` can produce the same results, I mean could this test in SQLQuerySuite be retained or just removed.

Reply:
It seems we can remove the test in `SQLQuerySuite`, may be I have a misunderstanding to `datetime.sql` and the usage.